lib/AST/ASTImporter.cpp

```diff
@@ -2575,9 +2584,9 @@ ExpectedDecl ASTNodeImporter::VisitRecordDecl(RecordDecl *D) {
     D2 = D2CXX;
     D2->setAccess(D->getAccess());
     D2->setLexicalDeclContext(LexicalDC);
-    if (D->getFriendObjectKind() == Decl::FOK_None &&
-        (!DCXX->getDescribedClassTemplate() || DCXX->isImplicit()))
-      LexicalDC->addDeclInternal(D2);
+    addDeclToDCIfVisible(D, D2, DC, LexicalDC);
```

Comment:
This is redundant with line 2655.
Can't we just have it called once? After both the CXXRecord and the simple Record is handled? Around line 2657?

Reply:
I wanted that the decl is added before other operations (mostly `Import`) are done.

Reply:
Okay, but does that really matter? I mean other `Imports` find the Decl through the `ASTImporterLookupTable` (unless LLDB).